### DISCUSSION TITLE
Migration Guide From Stability's StableSwarmUI to independent SwarmUI

### DISCUSSION THREAD

**Discussion by mcmonkey4eva** [mcmonkeyprojects/SwarmUI#2] [Announcements]
Here's a quick guide for how to migrate from Stability AI / StableSwarmUI, to the new independent SwarmUI

# Windows

First update by going to `Server` tab and clicking `Update and Restart`
![image](https://github.com/mcmonkeyprojects/SwarmUI/assets/4000772/eaf172d5-7e92-4b25-bf8e-9a1ad1ea1429)

Then open the folder and double-click "migrate-windows.bat"
![image](https://github.com/mcmonkeyprojects/SwarmUI/assets/4000772/01e66e59-2599-40c9-8136-e2f3a0ccf364)

Then launch swarm again, and update again (`Update and Restart`)

And, boom, you're done!

# Linux, Mac, or Manual on Windows

Open a terminal in the Swarm folder, and use commands:

`git remote set-url origin https://github.com/mcmonkeyprojects/SwarmUI`

then

`git pull`

If all went well, you should now be migrated. Easy as that!

----

Ps join the Discord https://discord.gg/q2y38cqjNw

**Comment by ArthriticOne** [comment]
Incredible work.  Thank you so much.  My flavor is Swarm.

**Comment by ionite34** [comment]
A note for [StabilityMatrix](https://github.com/LykosAI/StabilityMatrix) users, we've implemented automatic migration to SwarmUI since https://github.com/LykosAI/StabilityMatrix/releases/tag/v2.11.2

- Do not run `migrate-windows.bat` as it will have no effect.
- Please ensure your app is up to date, then click **Update** to migrate to this repo:

![image](https://github.com/mcmonkeyprojects/SwarmUI/assets/13956642/6989da4e-77e3-4961-816d-49da08e9a3ed)

**Reply by Blogwelt** [reply]
So nothing has been renamed in my Matrix version yet. I get one or two updates every day, but the name is still StableSwarmUI. If I look under Add Package, I now also find the SwarmUI there

**Comment by 25484** [comment]
The input area for prompts and reverse prompts is too small, and it doesn't support actively adjusting the area size. I hope it can be optimized.

**Comment by Jonadi12** [comment]
I have StabilityMatrix and used the dmg-file for the macos, but updating it make problems. 
![error-2](https://github.com/user-attachments/assets/d054630f-a30f-40b1-90db-4d320a54de3c)
"Could not connect to backend. The JSON value could not be converted to system.collections.generic-list' 1 [System.String]. Path $[1] LineNumber:0 BytePositionInLine: 261"
Hmmm... what can I do? :-)
